helm-charts/HPA.md

```diff
@@ -100,60 +100,37 @@ used model, data type and image versions, the specified resource values may need
+Above step created custom metrics config for Prometheus-adapter, suitable for HPA use.
 
-Take backup of existing custom metrics config before replacing it:
+However, Helm does not allow OPEA chart to overwrite Prometheus-adapter configMap, as
+it belongs to another chart. Therefore manual step is needed to overwrite its current
+custom metric rules.
 
-```console
-$ prom_ns=monitoring  # Prometheus/-adapter namespace
-$ name=$(kubectl -n $prom_ns get cm --selector app.kubernetes.io/name=prometheus-adapter -o name | cut -d/ -f2)
-$ kubectl -n $prom_ns get cm/$name -o yaml > adapter-config.yaml.bak
-```
-
-Save generated config with values matching current adapter config:
+Following will overwrite current Prometheus-adapter custom metric rules with OPEA Helm
+generated ones, and restart Prometheus-adapter to apply the new rules:
 
 ```console
-$ chart=chatqna  # OPEA chart release name
-$ kubectl get cm/$chart-custom-metrics -o yaml | sed \
-  -e "s/name:.*custom-metrics$/name: $name/" \
-  -e "s/namespace: default$/namespace: $prom_ns/" \
-  > adapter-config.yaml
+scripts/install-custom-metrics.sh monitoring chatqna
 ```
 
-NOTE: if there are existing custom metric rules you need to retain, add them from saved
-`adapter-config.yaml.bak` to `adapter-config.yaml` file now!
-
-Overwrite current Prometheus-adapter configMap with generated one:
-
-```console
-$ kubectl delete -n $prom_ns cm/$name
-$ kubectl apply -f adapter-config.yaml
-```
+(Prometheus-adapter is assumed to be in `monitoring` namespace, and new rules to be
+generated by a ChatQnA chart release named as `chatqna`.)
 
-And restart it, so that it will use the new config:
-
-```console
-$ selector=app.kubernetes.io/name=prometheus-adapter
-$ kubectl -n $prom_ns delete $(kubectl -n $prom_ns get pod --selector $selector -o name)
-```
+YAML backup of new and previous rules are saved to current directory.
 
 ## Verify
 
 After [verifying that service metrics work](monitoring.md#verify),
 one can verify that HPA rules can access custom metrics based on them.
 
-Verify that there are custom metrics from inferencing service(s), prefixed with the chart name:
+Verify that custom metric values are available for scaling the services:
 
 ```console
-$ kubectl get --raw /apis/custom.metrics.k8s.io/v1beta1 | jq .resources[].name
+watch -n 5 scale-monitor-helm.sh default chatqna
 ```
 
-And HPA rules have TARGET values for HPA controlled service deployments (instead of `<unknown>`):
-
-```console
-$ ns=default  # OPEA namespace
-$ kubectl -n $ns get hpa
-```
+(Assumes that HPA scaled chart is installed to `default` namespace with `chatqna` release name.)
 
 **NOTE**: inferencing services provide metrics only after they've processed their first request.
-And reranking service is used only after query context data has been uploaded!
+And reranking service is used only after query context data has been uploaded. Before that,
+you will not see any metrics!
```

Comment:
The reranking service is used only after the query context data has been uploaded. Until then, no metrics will be available.